apps/src/applab/dataBrowser/DataOverview.jsx

```diff
@@ -125,9 +81,28 @@ const DataOverview = React.createClass({
 });
 
 export default connect(state => ({
-  view: state.data.view
+  view: state.data.view,
+  tableListMap: state.data.tableListMap || {}
 }), dispatch => ({
   onViewChange(view, tableName) {
     dispatch(changeView(view, tableName));
+  },
+  onTableAdd(tableName) {
+    // Add the table name to the model even though it doesn't exist in the database yet,
+    // so that it appears in the overview even if we don't create a record in it.
+    // addTableName() is idempotent, so we won't double-count the table name later
+    // when the the first record is added to it.
+    dispatch(addTableName(tableName));
+
+    dispatch(changeView(DataView.TABLE, tableName));
+  },
+  onTableDelete(tableName) {
+    // Explicitly remove the table from the model to make sure it disappears from the
+    // list, since we won't receive a delete event from the database if the table did
+    // not exist in the database. This could happen when deleting a table after adding it
+    // without adding any records to it.
+    dispatch(deleteTableName(tableName));
+
+    FirebaseStorage.deleteTable(tableName);
```

Comment:
I agree that this doesn't belong here, however I'd prefer to keep it out of redux actions as well because "Actions are payloads of information that send data from your application to your store" http://redux.js.org/docs/basics/Actions.html . Instead I've made the React component call FirebaseStorage directly, outside of redux. This is similar to how we set and delete key/value pairs.